src/mlpack/tests/main_tests/range_search_test.cpp

```diff
@@ -0,0 +1,389 @@
+                          "hilbert-r", "r-plus", "r-plus-plus", "vp","rp",
+                          "max-rp", "ub", "oct"};
+
+  if (!data::Load("iris.csv", inputdata))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+  if (!data::Load("iris_test.csv", querydata))
+    BOOST_FAIL("Unable to load dataset iris_test.csv!");
+
+  //Define Base with kd Tree
+  SetInputParam("tree_type", trees[0]);
+  SetInputParam("min", minv);
+  SetInputParam("max", maxv);
+  SetInputParam("distances_file", distancefile);
+  SetInputParam("neighbors_file", neighborsfile);
+  SetInputParam("reference", inputdata);
+  SetInputParam("query", querydata);
+
+  mlpackMain();
+
+  neighbors = ReadData<size_t>(neighborsfile);
+  distances = ReadData<double>(distancefile);
+  RSModel* outputmodel1=CLI::GetParam<RSModel*>("output_model");
+
+  for (size_t i = 1;i < trees.size() ; i++)
+  {
+    if (!data::Load("iris.csv", inputdata))
+      BOOST_FAIL("Unable to load dataset iris.csv!");
+    if (!data::Load("iris_test.csv", querydata))
+      BOOST_FAIL("Unable to load dataset iris_test.csv!");
+
+    SetInputParam("min", minv);
+    SetInputParam("max", maxv);
+    SetInputParam("distances_file", distancefile);
+    SetInputParam("neighbors_file", neighborsfile);
+    SetInputParam("query", querydata);
+    SetInputParam("reference", inputdata);
+    SetInputParam("tree_type", trees[i]);
+
+    mlpackMain();
+
+    neighborstemp = ReadData<size_t>(neighborsfile);
+    distancestemp = ReadData<double>(distancefile);
+
+    CheckMatrices(neighbors, neighborstemp);
+    CheckMatrices(distances, distancestemp);
+    BOOST_REQUIRE_EQUAL(CheckModelSerial(outputmodel1,
+                                   CLI::GetParam<RSModel*>("output_model")), 0);
+  }
+}
+
+BOOST_AUTO_TEST_CASE(RandomBasisTesting)
+{
+  string distancefile = "distances.csv";
+  string neighborsfile = "neighbors.csv";
+  double minv = 0, maxv = 3;
+  arma::mat querydata, inputdata;
+  vector<vector<size_t>> neighbors, neighborstemp;
+  vector<vector<double>> distances, distancestemp;
+  if (!data::Load("iris.csv", inputdata))
+    BOOST_FAIL("Unable to load dataset iris.csv!");
+  if (!data::Load("iris_test.csv", querydata))
+    BOOST_FAIL("Unable to load dataset iris_test.csv!");
+
+  SetInputParam("min", minv);
+  SetInputParam("max", maxv);
+  SetInputParam("distances_file", distancefile);
+  SetInputParam("neighbors_file", neighborsfile);
+  SetInputParam("reference", inputdata);
+
+  mlpackMain();
+
+  RSModel* outputmodel = move(CLI::GetParam<RSModel*>("output_model"));
+
+  SetInputParam("min", minv);
+  SetInputParam("max", maxv);
+  SetInputParam("distances_file", distancefile);
+  SetInputParam("neighbors_file", neighborsfile);
+  SetInputParam("reference", inputdata);
+  SetInputParam("random_basis",true);
+
+  mlpackMain();
+
+  BOOST_REQUIRE_EQUAL(CheckModelSerial(outputmodel,
+                                    CLI::GetParam<RSModel*>("output_model")),0);
```

Comment:
With the changes I suggested to `CheckModelSerial()` you could change this to `BOOST_REQUIRE_NE(ModelToString(outputmodel), ModelToString(CLI::GetParam<RSModel*>("output_model")));`.